server/src/main/java/com/linecorp/centraldogma/server/internal/api/AdministrativeService.java

```diff
@@ -53,111 +51,47 @@ public AdministrativeService(CommandExecutor executor) {
-        //                   | true -> false | bad_request exception
-        //-------------------+---------------+-----------------------------------
-        //    true -> false  | true -> true  | setWritable(false)
-        //                   | true -> false | setWritable(false) & stop executor.
-        //-------------------+---------------+-----------------------------------
-        //    false -> true  | true -> true  | setWritable(true)
-        //                   | false -> true | setWritable(true) & start executor.
-        //-------------------+---------------+-----------------------------------
-        //    false -> false | true -> false | Stop executor.
-        //                   | false -> true | Start executor.
-        final boolean writable = writableNode.asBoolean();
-        final boolean replicating = replicatingNode.asBoolean();
-        if (writable && !replicating) {
-            return HttpApiUtil.throwResponse(ctx, HttpStatus.BAD_REQUEST,
-                                             "'replicating' must be 'true' if 'writable' is 'true'.");
-        }
-        if (oldStatus.writable == writable && oldStatus.replicating == replicating) {
-            throw HttpStatusException.of(HttpStatus.NOT_MODIFIED);
-        }
-
-        if (oldStatus.writable != writable) {
-            executor().setWritable(writable);
-            if (writable) {
-                logger.warn("Left read-only mode.");
-            } else {
-                logger.warn("Entered read-only mode. replication: {}", replicating);
+        final ServerStatus newStatus = statusRequest.serverStatus();
+        if (statusRequest.scope() == Scope.LOCAL) {
+            // Validate the new status for the local scope. Other servers may have different status.
+            if (oldStatus == newStatus) {
+                throw HttpStatusException.of(HttpStatus.NOT_MODIFIED);
             }
-        }
 
-        if (oldStatus.replicating != replicating) {
-            if (replicating) {
-                return executor().start().handle((unused, cause) -> {
-                    if (cause != null) {
-                        logger.warn("Failed to start the command executor:", cause);
-                    } else {
-                        logger.info("Enabled replication. read-only: {}", !writable);
-                    }
-                    return status();
-                });
-            }
-            return executor().stop().handle((unused, cause) -> {
-                if (cause != null) {
-                    logger.warn("Failed to stop the command executor:", cause);
-                } else {
-                    logger.info("Disabled replication");
-                }
+            return CompletableFuture.supplyAsync(() -> {
+                executor().statusManager().setWritable(newStatus.writable());
+                executor().statusManager().setReplicating(newStatus.replicating());
+                statusManager.updateStatus(newStatus);
```

Comment:
In administrativeService we do:
```java
serverStatusManager.updateStatus(c.serverStatus());
statusManager().updateStatus(c);
```
However, we update `CommandExecutorStatusManager` first. Could you apply them in the same order?

Reply:
Sorry, my previous comment was unclear. 😓 
We do in `StandaloneCommandExecutor`:
```java
return CompletableFuture.supplyAsync(() -> {
    serverStatusManager.updateStatus(c.serverStatus());
    statusManager().updateStatus(c);
```
However, we do it here:
```java
return CompletableFuture.supplyAsync(() -> {
    executor().statusManager().updateStatus(newStatus);
    serverStatusManager.updateStatus(newStatus);
```
Shouldn't we do the same order?